.circleci/soltest_all.sh

```diff
@@ -71,8 +52,11 @@ do
         [ "${EVM}" = "byzantium" ] && [ "${OPTIMIZE}" = "0" ] && EWASM_ARGS="--ewasm"
         ENFORCE_GAS_ARGS=""
         [ "${EVM}" = "istanbul" ] && ENFORCE_GAS_ARGS="--enforce-gas-cost"
+        # Run SMTChecker tests only when OPTIMIZE == 0
+        DISABLE_SMTCHECKER=""
+        [ "${OPTIMIZE}" == "0" ] && DISABLE_SMTCHECKER="-t !smtCheckerTests"
 
-        [[ " $RUN_STEPS " == *" $STEP "* ]] && EVM="$EVM" OPTIMIZE="$OPTIMIZE" SOLTEST_FLAGS="$SOLTEST_FLAGS $ENFORCE_GAS_ARGS $EWASM_ARGS" BOOST_TEST_ARGS="-t !@nooptions" "${REPODIR}/.circleci/soltest.sh"
+        [[ " $RUN_STEPS " == *" $STEP "* ]] && EVM="$EVM" OPTIMIZE="$OPTIMIZE" SOLTEST_FLAGS="$SOLTEST_FLAGS $ENFORCE_GAS_ARGS $EWASM_ARGS" BOOST_TEST_ARGS="-t !@nooptions $DISABLE_SMTCHECKER" "${REPODIR}/.circleci/soltest.sh"
```

Comment:
@cameel I'm not sure here, do you think the `@nooptions` flag is still needed?

Reply:
Not sure. I think it was introduced specifically to make this a more general mechanism (instead of simply disablng the SMT checker, see https://github.com/ethereum/solidity/pull/9359/files#r451799902)) so I'd say it should stay.